docs/topics/authentication.rst

```diff
@@ -0,0 +1,15 @@
+======================
+Authentication Systems
+======================
+
+This system has migrated from the older version management library (msrestazure) to 
```

Comment:
**suggestion (blocking):** Think of this document from more of a user perspective. We want to answer their questions when they come to the documentation.

What are the different login mechanisms that we support for users? Is there an order to the authentication types that is respected? How would a user get Salt to recognize each mechanism?

The service principal workflow isn't well-documented because it seems that MS wants to move away from it (maybe?). However, we should be able to say _something_ about how a user might pass credentials that is userful.

The [username/password](https://docs.microsoft.com/en-us/azure/developer/python/sdk/authentication-additional-methods#authentication-with-a-username-and-password) workflow has a little documentation, but it's mostly in the context of developers. The second paragraph is pretty good though, because it gets across that this mechanism is not very secure. Aside from that, what information would someone need and how would they provide it to Salt?

Finally, the ["Default" credential handler](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/identity/azure-identity#defaultazurecredential) has a certain list of mechanisms that it supports. Users don't really care what it's called, but they need to know the different ways that it supports their authentication workflows. At minimum, we'd want to replicate that list.

If you can wrap all of that information in a narrative, that will make the documentation more readable... and therefore more usable.